app/models/notifications/mentioned_in_comment.rb

```diff
@@ -10,8 +10,15 @@ def deleted_translation_key
       "notifications.mentioned_in_comment_deleted"
     end
 
-    def self.filter_mentions(mentions, _mentionable, _recipient_user_ids)
-      mentions
+    def self.filter_mentions(mentions, mentionable, _recipient_user_ids)
+      ppl = mentionable.people_allowed_to_be_mentioned
+      if ppl == "all"
+        mentions
+      else
+        mentions.to_ary.delete_if {|mention|
```

Comment:
Yes, I was also unsure about this, but I talked with @CSammy about that, and he meant that the DB should manage it, because there are other `WHERE` conditions to minimize the result set. 

I did a few tests locally now, and yes, it gets slower if you add more IDs to the `IN`, but 10000 are still ~10ms, so it should be "ok" (not really slow). Also I don't expect it to be more than 100, because I only know public posts with really many participants, private posts are mostly less active (at least the private posts I know about).

But I'm still unsure ... I thought the `where` would be more readable, but if you prefer the `delete_if` I'm also fine with it. But maybe `keep_if` is more intuitive then?